OneCommunityGlobal/HighestGoodNetworkApp
# Fixing all button to expand all tasks

### **Description**
--------------------------------------------------------------

This PR is regarding issue # 7  - "All" button doesn't open all folders in the WBS. This issues are listed below what the user encounter.

- Dashboard→ Other Links → Projects → AAA USE THIS PROJECT → WBS Icon → Jae's WBS - USE THIS → click the “All” button → then click and reclick some of the top open folders and you’ll see they aren’t actually opening to show their full contents

- I’m also having issues on the Main where I can’t open a folder without using the “all” button. Can’t reproduce on the Dev though… so hoping fixing “a” above will fix this too. 


### **Related PRS (if any):**
--------------------------------------------------------------
None

### **Mainly changes explained:**
--------------------------------------------------------------

1. update the open folder state to reflect the changes using useEffect hooks
2. split the fetching of tasks into 2 states. (fetch only the parent task to reduce the latency and the second is to load all tasks which takes longer)
3. adding loading spinner to improve the UI while loading

### How to test:
--------------------------------------------------------------
Check into the current branch
Do npm install and ... to run this PR locally
Log in as an admin user
Dashboard→ Other Links → Projects → AAA USE THIS PROJECT → WBS Icon → Jae's WBS - USE THIS → click the “Expand All” button 
check all the tasks are being displayed correctly
click "collapse All" button to see the parent tasks are being display

### Screenshots or videos of changes:
-----------------------------------------------------------------

https://user-images.githubusercontent.com/18197013/227554825-7e54def6-9351-4801-89a7-9f22c798e822.mov
